packages/babel-parser/src/parser/expression.js

```diff
@@ -2088,7 +2103,10 @@ export default class ExpressionParser extends LValParser {
     allowPlaceholder: ?boolean,
   ): ?N.Expression {
     let elt;
-    if (allowEmpty && this.match(tt.comma)) {
+    if (this.match(tt.comma)) {
+      if (!allowEmpty) {
+        this.raise(this.state.pos, Errors.UnexpectedToken, ",");
```

Comment:
I resort to the most generic error message. I still feel awkward mentioning holes when throwing `f(,)`. Any suggestions are definitely welcome.

At least the error is now recoverable.